src/FxCop/Desktop.Analyzers/Core/Security/CA3075DiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,817 @@
+
+        // Do not explicitly set XmlDocument.XmlResolver property to an insecure value (anything thing other than null or a instance of XmlSecureResolver).
+        // This DOES NOT trigger if the XmlDocument instance is created but its XmlResolver property is never explicitly set
+        // Partially match FxCop CA3053 CA3053 : XmlDocumentWithNoSecureResolver
+        internal static DiagnosticDescriptor RuleXmlDocumentSetInsecureResolver = CreateDiagnosticDescriptor(DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlDocumentWithNoSecureResolverDiagnosis)),
+                                                                                                              DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlDocumentUseSecureResolverDescription)));
+
+        // if XmlTextReader is constructed in the method/class as field, must set entity and resource resolution related properties to secure values
+        // (using object initialier list in case of field), or use XmlReader.Create() instead
+        // i.e. DtdProcessing == Dtdprocessing.Prohibit/Ignore && XmlResolver == XmlSecureResolver/null (no warning if so)    
+        // Doesn't directly match any FxCop rule, related to:
+        //      FxCop CA 3054
+        //      FxCop CA 3069
+        internal static DiagnosticDescriptor RuleXmlTextReaderConstructedWithNoSecureResolution = CreateDiagnosticDescriptor(DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlTextReaderConstructedWithNoSecureResolutionDiagnosis)),
+                                                                                                                             DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlTextReaderInsecureResolutionDescription)));
+
+        // if XmlTextReader instance is passed in as parameter (or is a field), Do not set entity or resource resolution related properties to insecure values
+        // i.e. DtdProcessing == Dtdprocessing.Parse || XmlResolver != XmlSecureResolver/null     
+        // Doesn't directly match any FxCop rule, related to:
+        //      FxCop CA 3054
+        //      FxCop CA 3069
+        internal static DiagnosticDescriptor RuleXmlTextReaderSetInsecureResolution = CreateDiagnosticDescriptor(DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlTextReaderSetInsecureResolutionDiagnosis)),
+                                                                                                                 DiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.XmlTextReaderInsecureResolutionDescription)));
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> supportedDiagnostics =ImmutableArray.Create(RuleDoNotUseUnsafeDtdProcessingOverloads,
```

Comment:
There's one id with all these descriptors with different titles and descriptions? We shouldn't do that as discussed in the previous PR.

I'm curious why all these rules have been combined into one id? They don't seem to be about one thing (which was true for the "do not use x algorithm" rule.